### PR TITLE
Add rapidcheck 20210702

### LINKED ISSUE #7026: [request] rapidcheck/20210702
### Package Details
  * Package Name/Version: **rapidcheck/20210702**

The above-mentioned version solves issue #5205 and is not yet available as a recipe. Please add this version.

### PR DESCRIPTION
**Pull request by dmn-star**
Specify library name and version:  **rapidcheck /20210702**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

Closes  #7026
---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/rapidcheck/all/conandata.yml [hunk 1]
```diff
@@ -5,3 +5,6 @@ sources:
   "20210107":
     url: "https://github.com/emil-e/rapidcheck/archive/37bcab858dfc398634d409c973363088703ba96e.zip"
     sha256: "35433d0417fb945abe2b56b720b03d6d92da1188e760c50662c243dac42a6717"
+  "20210702":
```
Comment (madebr) [inline recipes/rapidcheck/all/conandata.yml:8]: Please add a `cci.` prefix to all packages.
I think the approach of https://github.com/conan-io/conan-center-index/pull/6951 can be re-used here.

Comment (jgsogo) [reply recipes/rapidcheck/all/conandata.yml:8]: Better if we create a recipe-revision for _old_ versions with a message to inform consumer about the new reference they should start using: https://github.com/dmn-star/conan-center-index/pull/1

Comment (dmn-star) [reply recipes/rapidcheck/all/conandata.yml:8]: Wouldn't it be possible to simply delete the old revision in that case?

Comment (jgsogo) [reply recipes/rapidcheck/all/conandata.yml:8]: The _issue_ is that anyone using `rapidcheck.20210107` will keep using it _forever_. The center will return the latest recipe-revision available for that reference. A new recipe-revision for a different reference (even if it is the same library name) doesn't invalidate other versions.

The path here is to create a new recipe revision (that will become the latest one) with the warning/error message, people trying to update `rapidcheck.20210107` will get the error and can decide to move to the `.cci` version and new recipe revisions.

Comment (jgsogo) [reply recipes/rapidcheck/all/conandata.yml:8]: > Wouldn't it be possible to simply delete the old revision in that case?

Sorry, I read _version_ not revision. No, this is not possible. We don't remove anything from ConanCenter.

Comment (dmn-star) [reply recipes/rapidcheck/all/conandata.yml:8]: ok

**Review by jgsogo — APPROVED** [review]
Thanks! A PR after this one removing the old versions would be very welcome 😃

**Comment by SSE4** [comment]
around here: https://github.com/conan-io/conan-center-index/blob/dd1c0a771fb30d606faa6a44080b288da0a2b940/recipes/rapidcheck/all/conanfile.py#L113
need something like:
```
version = self.version
if version.startswith("cci."):
    version = version[4:]
```

**Comment by dmn-star** [comment]
Will something like that work too? 
tools.Version(self.version) < "cci.20201218"

**Comment by dmn-star** [comment]
https://github.com/conan-io/conan/issues/9473

**Review by prince-chrismc — APPROVED** [review]

**Review by SSE4 — APPROVED** [review]

**Review by jgsogo — APPROVED** [review]

**Pull request merged**